src/clients/c/tb_client/context.zig

```diff
@@ -301,19 +297,14 @@ pub fn ContextType(
             packet: *Packet,
             result: PacketError![]const u8,
         ) void {
-            self.messages_available += 1;
-            assert(self.messages_available <= constants.client_request_queue_max);
+            assert(self.client.messages_available <= constants.client_request_queue_max);
 
             // Signal to resume sending requests that was waiting for available messages.
-            if (self.messages_available == 1) self.thread.signal.notify();
+            if (self.client.messages_available == 1) self.thread.signal.notify();
```

Comment:
This is subtle. In  the `Client.zig`, I actaully allow the users to retain messages beyond request-response itself. Eg, the user can do

```
const message = client.get_message();
some_global_var.message = message.ref();
client.request(message...)
```

So a request might become available either when a reply arrives, or when the user drops their outstanding reference. If the latter case happened here, that would have been a lost wakeup. 

However, tb_client fully controlls message and it doesn't retain references elsewher.